### DISCUSSION TITLE
Improve .ppignore functionality

### DISCUSSION THREAD

**Discussion by d2r2** [photoprism/photoprism#1043] [Ideas]
`.ppignore` is a powerful customization tool which very helpful if you already have big picture archive collected over the years.

But still I have the proposal to improve it. For instance, my regular photo-set contains raw images (jpeg and raw files) in the folder and subfolder in it named like "_processed" or "_selected" where processed jpeg files are located. In this workflow any attempt to expose to Photoprism files located not in the folder but in the subfolder are unsuccessful. Because, if I create `.ppignore` file with `*.jpeg`/`*.*` mask it will also ignore jpeg files in any nested subfolder.

So my proposal is aimed to extend `.ppignore` functionality in this way. I suppose it could be various technical decisions for this, but one I have in mind - introduce `.ppignorelocal` file, which have absolutely same syntax but acts only for current folder and do not spread out to subfolders.

I can try to implement pull request for this, if community accept proposed improvement.
If you have concerns about how this should be implemented, then I am also ready to discuss it.

**Comment by alexisLefebvre** [comment]
Idea: treat a pattern starting with `/` as relative to the folder where the `.ppignore` is placed. So `/*.jpg` would match with the files in the same directory that the `.ppignore` but not in the sub-folders.

**Reply by lastzero** [reply]
Providing an full path should already work... not sure if it works directly with / though. In the worst case, put the .ppignore file in the parent folder.

**Reply by d2r2** [reply]
> Idea: treat a pattern starting with `/` as relative to the folder where the `.ppignore` is placed. So `/*.jpg` would match with the files in the same directory that the `.ppignore` but not in the sub-folders.

Agree that it will be good with mask `/*.jpg`, but according to my test I have completed right now, it doesn't work.

**Reply by lastzero** [reply]
Did you try putting the ignore file in the parent folder and use `child/*.jpg`?

**Reply by d2r2** [reply]
> Providing an full path should already work... not sure if it works directly with / though. In the worst case, put the .ppignore file in the parent folder.

Full path as I see works, but there was an idea to have unified `.ppignore` file for such cases, rather that modify file with content `/photoprism/originals/Denis/Kemer'07'2010/*.jpg` for every occurrence.

**Reply by lastzero** [reply]
Might come back to this later as tons of pull requests and feature requests are waiting for us...

**Reply by d2r2** [reply]
OK, I see.

**Reply by JamesSwift** [reply]
> Idea: treat a pattern starting with `/` as relative to the folder where the `.ppignore` is placed. So `/*.jpg` would match with the files in the same directory that the `.ppignore` but not in the sub-folders.

Isn't it more standardized for paths starting with `/` to be absolute, and paths starting with `./` to be relative to the current directory?

**Reply by alexisLefebvre** [reply]
> Isn't it more standardized for paths starting with `/` to be absolute, and paths starting with `./` to be relative to the current directory?

My proposal was to use that pattern as relative to the `.ppignore` file, but you're right it would make it unusable to match a full path.
